hivemq-edge/src/test/java/com/hivemq/embedded/internal/EmbeddedHiveMQImplHttpsTest.java

```diff
@@ -0,0 +1,186 @@
+    @Rule
+    public @NotNull TemporaryFolder tmp = new TemporaryFolder();
+
+    private @NotNull File data;
+    private @NotNull File license;
+    private @NotNull File extensions;
+    private @NotNull File conf;
+    private int randomPort;
+    private int randomApiPort;
+
+    @Before
+    public void setUp() throws Exception {
+        data = tmp.newFolder("data");
+        license = tmp.newFolder("license");
+        extensions = tmp.newFolder("extensions");
+        conf = tmp.newFolder("conf");
+        randomPort = RandomPortGenerator.get();
+        randomApiPort = RandomPortGenerator.get();
+
+        final @NotNull File keystoreFile = new File("src/test/resources/keystores/hivemqtest.jks");
+        assertTrue(keystoreFile.exists(), "The test keystore is not found");
+
+        final @NotNull String configXmlString = "" +
+                "<hivemq>\n" +
+                "    <mqtt-listeners>\n" +
+                "        <tcp-listener>\n" +
+                "            <port>" +
+                randomPort +
+                "</port>\n" +
+                "            <bind-address>0.0.0.0</bind-address>\n" +
+                "        </tcp-listener>\n" +
+                "    </mqtt-listeners>\n" +
+                "    <admin-api>\n" +
+                "        <enabled>true</enabled>\n" +
+                "        <listeners>\n" +
+                "            <https-listener>\n" +
+                "                <port>" +
+                randomApiPort +
+                "</port>\n" +
+                "                <bind-address>0.0.0.0</bind-address>\n" +
+                "                <tls>\n" +
+                "                    <keystore>\n" +
+                "                        <path>" +
+                keystoreFile.getAbsolutePath() +
+                "</path>\n" +
+                "                        <password>12345678</password>\n" +
+                "                        <private-key-password>12345678</private-key-password>\n" +
+                "                    </keystore>\n" +
+                "                </tls>\n" +
+                "            </https-listener>\n" +
+                "        </listeners>\n" +
+                "    </admin-api>\n" +
+                "</hivemq>";
+        FileUtils.write(new File(conf, "config.xml"), configXmlString, StandardCharsets.UTF_8);
+    }
+
+    @Test(timeout = 20000L)
+    public void embeddedHiveMQ_whenHttpsEnabled_readsConfig() throws Exception {
+        try (final EmbeddedHiveMQImpl embeddedHiveMQ = new EmbeddedHiveMQImpl(conf, data, extensions, license)) {
+            embeddedHiveMQ.start().join();
+
+            final ConfigurationService configurationService = embeddedHiveMQ.bootstrapConfig();
+
+            final List<Listener> mqttListeners = configurationService.listenerConfiguration().getListeners();
+            assertEquals(1, mqttListeners.size(), "The listener count should be 1");
```

Comment:
Sorry for not pointing it out.
For asserts we stick to AssertJ.
You will still see other assert styles in older code but the general rule of thumb is to go with AssertJ for new stuff.

Reply:
Oh, thank you for the reminder. I'll adjust it.